Java/ds/SparseMatrices.java

```diff
@@ -0,0 +1,120 @@
+//          SPARSE MATRIX
+
+
+//  Java program for Sparse Matrix Representation using Array 
+import java.util.*;
+
+public class SparseMatrices{
+
+    public static void main(String[] args) {
+        // Varibale Declaration
+        int rows;
+        int columns;
+        // Matrix Variable to store the input from the user
+        int inputMatrix[][] = new int[20][20];
+        int sparseMatrix[][] = new int[20][3];
+        int i;
+        int j;
+        int k;
+        int count = 0;
+
+        // Creating Scanner Object to read input from user
+        Scanner s = new Scanner(System.in);
+
+        System.out.println("Enter the number of rows of the matrix");
+        // Reading the number of rows of the matrix from the user
+        rows = s.nextInt();
+
+        System.out.println("Enter the number of columns of the matrix");
+        // Reading the number of columns of the matrix from the user
+        columns = s.nextInt();
+
+        System.out.println("Enter the elements of the matrix");
+        for (i = 0; i < rows; i++) {
+            for (j = 0; j < columns; j++) { // Reading the elements of the matrix from the user
+                inputMatrix[i][j] = s.nextInt();
+            }
+        }
+        // Displaying the input matrix
+        System.out.println("Displaying the elements of given matrix");
+        for (i = 0; i < rows; i++) {
+            for (j = 0; j < columns; j++) {
+                System.out.print(inputMatrix[i][j] + " ");
+            }
+            // printing a new line after each row of the matrix
+            System.out.println();
+        }
+
+        // First row's first column denotes the number of rows of a matrix in sparse
+        // matrix
+        sparseMatrix[0][0] = rows;
+        // First row's second column denotes the number of columns of a matrix in sparse
+        // matrix
+        sparseMatrix[0][1] = columns;
+        k = 1;
+        for (i = 0; i < rows; i++) {
+            for (j = 0; j < columns; j++) { // If a particular input of matrix is non-zero
+                if (inputMatrix[i][j] != 0) { // Storing the row index of that particular non zero number
+                    sparseMatrix[k][0] = i;
+                    // Storing the column index of that particular non zero number
+                    sparseMatrix[k][1] = j;
+                    sparseMatrix[k][2] = inputMatrix[i][j];
+                    k++;
+                    // count variable keeps a track on non-zero entries in the given matri
+                    count += 1;
+                }
+            }
+            // First row's third column denotes
+            // number of non - zero elements in
+            // sparseMatrix
```

Comment:
Use Multi-Line Comments

Reply:
I have changed it into Multi-Line Comments